api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java

```diff
@@ -152,4 +263,91 @@ public void fail(@NotNull EconomyException exception) {
      */
     void deleteAccount(@NotNull EconomySubscriber<Boolean> subscription);
 
+    /**
+     * Returns the {@link Currency Currencies} this {@code Account} holds balance for.
+     *
+     * @param subscription the {@link EconomySubscriber} accepting the currencies
+     * @author MrIvanPlays
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    void retrieveHeldCurrencies(@NotNull EconomySubscriber<Collection<UUID>> subscription);
+
+    /**
+     * Request the {@link EconomyTransaction} history, limited by the {@code transactionCount}, of this {@code Account}.
+     *
+     * <p>If the specified {@code transactionCount} is higher than the known transactions, then this will return all the
+     * transactions.
+     * <p>If the specified {@code transactionCount} is {@code -1}, then this will return all the transactions. <i>If you
+     * really need all the transactions, please use this with care, because an economy provider might do a query to a database
+     * .</i>
+     *
+     * @param transactionCount the count of the transactions wanted
+     * @param subscription     the {@link EconomySubscriber} accepting the transaction history
+     * @author MrIvanPlays
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    void retrieveTransactionHistory(
```

Comment:
I really don't like the idea of potentially thousands of transactions being loaded, even if it isn't done on the main thread. Were I to implement an economy plugin I might offer a Stream of transactions that would request additional transactions as necessary to complete its history, but I would not ever support loading all transactions in one single bucket under any circumstances - I'd cap this somewhere in the 25-100 range and force you to use the timestamp-based options if you wanted more.

I would vastly prefer to see an overload `retrieveTransactionHistory(int transactionCount, @NotNull Temporal from, @NotNull Temporal to, @NotNull EconomySubscriber<Collection<EconomyTransaction>> subscription)`.
This would then become a default implementation:
```java
default void retrieveTransactionHistory(
    int transactionCount, @NotNull EconomySubscriber<Collection<EconomyTransaction>> subscription
) {
  retrieveTransactionHistory(transactionCount, NOW, EPOCH_START, subscription);
}
```